pipeline/stage03_trigger_detection/scripts/minima.py

```diff
@@ -6,17 +6,41 @@
 from utils import load_neo, write_neo, remove_annotations
 
 
-def detect_minima(asig, order):
+def detect_minima(asig, order, interpolation_points, sampling_time, interpolation):
     signal = asig.as_array()
+    
     t_idx, channel_idx = argrelmin(signal, order=order, axis=0)
 
-    sort_idx = np.argsort(t_idx)
+    if int(interpolation) == 0:
+        fitted_idx = []
+        fitted_idx_times = []
 
-    evt = neo.Event(times=asig.times[t_idx[sort_idx]],
-                     labels=['UP'] * len(t_idx),
-                     name='Transitions',
-                     minima_order=order,
-                     array_annotations={'channels':channel_idx[sort_idx]})
+        for i in range(0,len(t_idx)):
+        
+            if t_idx[i] - int((interpolation_points-1)/2) >= 0 and t_idx[i] + int((interpolation_points-1)/2)+1 <= len(signal[:,0]):
+                start = t_idx[i] - int((interpolation_points-1)/2)
+                end =  t_idx[i] + int((interpolation_points-1)/2)+1
+
+                X = list(range(start, end))
+                params = np.polyfit(X, signal[start:end, channel_idx[i]], 2)
+                minimum = -(1.*params[1])/(2.*params[0])
+                if minimum > 0:
+                    fitted_idx.append(minimum)
+                    fitted_idx_times.append(minimum*sampling_time)
+            
```

Comment:
else statement missing! fitted indexes and channel_idx must remain the same length